qiskit/pulse/instructions/instruction.py

```diff
@@ -267,3 +257,79 @@ def __repr__(self) -> str:
         return "{}({}{})".format(
             self.__class__.__name__, operands, f", name='{self.name}'" if self.name else ""
         )
+
+
+class FrameInstruction(Instruction, ABC):
+    """Instructions acting on a ``Frame`` (or ``MixedFrame``).
+
+
+    ``FrameInstruction``s can be defined on either a ``MixedFrame`` (=``Channel``) or a ``Frame``.
+    For a ``MixedFrame`` the instruction applies only to the specific ``MixedFrame``. However, for
+    a ``Frame`` the instruction will be broadcasted to all ``MixedFrame``s associated with that
+    ``Frame``.
+
+    ``FrameInstruction`` should be considered abstract, and should not be instantiated as is.
+    """
+
+    def _validate_and_format_frame(
+        self, target: PulseTarget, frame: Frame, mixed_frame: MixedFrame, channel: Channel
+    ) -> Union[Frame, MixedFrame, PulseChannel]:
+        """Called after initialization to validate instruction data.
+
+        Arguments:
+            target: The pulse target used in conjunction with ``frame`` to define the mixed frame
+                the instruction applies to.
+            frame: The frame the instruction applies to.
+            mixed_frame: The mixed_frame the instruction applies to.
+            channel: The channel the instruction applies to.
+
+        Returns:
+            Union[Frame, MixedFrame, PulseChannel]: The frame or mixed frame the instruction applies to.
+
+        Raises:
+            PulseError: If the combination of ``target``, ``frame``, ``mixed_frame`` and
+                ``channel`` doesn't specify a unique ``MixedFrame`` or ``Frame``.
+        """
+
+        if (frame is None) and (target is not None):
+            raise PulseError("target can not be provided without frame")
+        if (channel is not None) + (mixed_frame is not None) + (frame is not None) != 1:
+            raise PulseError("Exactly one of mixed_frame, channel or frame must be provided")
+
+        if target is not None:
+            if not isinstance(target, PulseTarget):
+                raise PulseError(f"Expected a PulseTarget, got {target} instead.")
+            if not isinstance(frame, Frame):
+                raise PulseError(f"Expected a Frame, got {frame} instead.")
+            inst_target = MixedFrame(target, frame)
+        else:
+            inst_target = mixed_frame or channel or frame
+
+        if not isinstance(inst_target, (PulseChannel, MixedFrame, Frame)):
+            raise PulseError(f"Expected a MixedFrame, Channel or Frame, got {inst_target} instead.")
+
+        return inst_target
+
+    @property
+    def channel(self) -> Union[PulseChannel, None]:
```

Comment:
This isn't really an addition. It's just a way to keep the `channel` property of all frame instructions with less code clutter. It will make it easier to deprecate it later.

Reply:
I would add channel property to each subclass rather in the baseclass. In this way newly added instructions will not gain legacy property. Anyways this is small difference and doesn't block merge.